.github/pull_request_template.md

```diff
@@ -1,61 +1,36 @@
-
-Before opening a PR, please note these guidelines:
-
-- Each PR should only address ONE topic and have an associated GitHub issue
-- No hardcoded or paths to personal directories should be present
-- No temporary or backup files should be committed
-- Any code that was disabled by being commented out should be removed
-- Please provide a on-line, but descriptive title to the PR e.g. Adds XYZ functionality or Fixes ABC bugs, etc.  DO NOT use branch names as titles.
--->
 
-**Description**
+INSTRUCTIONS
 
-<!-- Please include relevant motivation and context. -->
-<!-- Please include a summary of the change and which issue is fixed. -->
-<!-- List any dependencies that are required for this change. -->
+- Please do not commit temporary, backup, or binary files.
+- Please remove commented-out code.
+- Please ensure code, config files, etc., contain no hardcoded paths.
+- Please format code snippets in PR description/comments with ```code block``` or `inline code`.
+- Please consider adding your own review comments to guide other reviewers.
+
+-->
 
-<!-- Please provide reference to the issue this pull request is addressing. -->
-<!-- For e.g. Fixes #IssueNumber -->
+**Description**
 
-**Type of change**
+<!-- A summary of the change, including relevant motivation, context, useful links, etc. -->
 
-Please delete options that are not relevant.
+**Type**
 
-- [ ] Bug fix (non-breaking change which fixes an issue)
-- [ ] New feature (non-breaking change which adds functionality)
-- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-- [ ] This change requires a documentation update
+Select one or more:
 
-**How Has This Been Tested?**
+- [ ] Bug fix (corrects a known issue)
+- [ ] Code maintenance (refactoring, etc.)
+- [ ] New feature (adds functionality)
 
-<!-- Please describe the tests that you ran to verify your changes. -->
-<!-- Provide instructions so we can reproduce. -->
-<!-- Please also list any relevant details for your test configuration -->
+**Impact**
 
-<!-- Use the following as a guide to list your tests and delete options that are not relevant. Expand as necessary. -->
-<!--
-- [ ] pytests in GitHub actions.
-- [ ] Tests on XYZ OS/HPC/CSP
--->
+Select one:
 
-**Checklist**
+- [ ] This is a non-breaking change (existing functionality continues to work as expected)
+- [ ] This is a breaking change (changes existing functionality)
 
-- [ ] My code follows the style guidelines of this project
-- [ ] I have performed a self-review of my own code
-- [ ] I have commented my code, particularly in hard-to-understand areas
-- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
-- [ ] My changes generate no new warnings
-- [ ] New and existing tests pass with my changes
-- [ ] Any dependent changes have been merged and published
```

Comment:
- Some of this is enforced by code-quality checks at this point.
- Style guidelines -- at least those that can't be enforced by by automated tools -- are more, IMO, a subject for peer review, as style is so subjective and case-specific.
- Documentation requirements has its own sections now.
- I've never understood the "dependent changes" item.

Reply:
I assumed dependent changes meant task A is merged before task B because it is needed to pass the current test cases, etc. This would make sense if test coverage wasn't 100%. I agree with your ideas of removing these since they are covered in other checks now.